esp-riscv-rt/src/lib.rs

```diff
@@ -641,4 +825,139 @@ _vector_table:
 
 .option pop
 "#,
+#[cfg(any(feature="direct-vectoring", feature="direct-vectoring-esp32c6"))]
+r#"
+#this is required for the linking step, these symbols for in-use interrupts should always be overwritten by the user.
+.section .trap, "ax"
+.weak cpu_int_1_handler
+.weak cpu_int_2_handler
+.weak cpu_int_3_handler
+.weak cpu_int_4_handler
+.weak cpu_int_5_handler
+.weak cpu_int_6_handler
+.weak cpu_int_7_handler
+.weak cpu_int_8_handler
+.weak cpu_int_9_handler
+.weak cpu_int_10_handler
+.weak cpu_int_11_handler
+.weak cpu_int_12_handler
+.weak cpu_int_13_handler
+.weak cpu_int_14_handler
+.weak cpu_int_15_handler
+.weak cpu_int_16_handler
+.weak cpu_int_17_handler
+.weak cpu_int_18_handler
+.weak cpu_int_19_handler
+.weak cpu_int_20_handler
+.weak cpu_int_21_handler
+.weak cpu_int_22_handler
+.weak cpu_int_23_handler
+.weak cpu_int_24_handler
+.weak cpu_int_25_handler
+.weak cpu_int_26_handler
+.weak cpu_int_27_handler
+.weak cpu_int_28_handler
+.weak cpu_int_29_handler
+.weak cpu_int_30_handler
+.weak cpu_int_31_handler
+cpu_int_1_handler:
+cpu_int_2_handler:
+cpu_int_3_handler:
+cpu_int_4_handler:
+cpu_int_5_handler:
+cpu_int_6_handler:
+cpu_int_7_handler:
+cpu_int_8_handler:
+cpu_int_9_handler:
+cpu_int_10_handler:
+cpu_int_11_handler:
+cpu_int_12_handler:
+cpu_int_13_handler:
+cpu_int_14_handler:
+cpu_int_15_handler:
+cpu_int_16_handler:
+cpu_int_17_handler:
+cpu_int_18_handler:
+cpu_int_19_handler:
+cpu_int_20_handler:
+cpu_int_21_handler:
+cpu_int_22_handler:
+cpu_int_23_handler:
+cpu_int_24_handler:
+cpu_int_25_handler:
+cpu_int_26_handler:
+cpu_int_27_handler:
+cpu_int_28_handler:
+cpu_int_29_handler:
+cpu_int_30_handler:
+cpu_int_31_handler:
+    la ra, cpu_int_1_handler #panic spin since proper handler is not defined, this could also just load the default _start_trap_rust_hal address and let the hal handle it.
+    jr ra
+"#,
+}
+#[cfg(feature = "direct-vectoring-esp32c6")]
```

Comment:
Could we change this to `#[cfg(all(feature = "direct-vectoring", esp32c6)]`. I'd rather there was only one feature for  enabling `direct-vectoring`.

Reply:
In fact, we can change this to `#[cfg(all(feature = "direct-vectoring", plic)]` and get support for the esp32h2 as well.

Reply:
@onsdagens if we make `_handle_priority` and `_restore_priority` `extern` is esp-hal, we won't need to define them in esp-riscv-rt, right?

Reply:
That's true, this also allows us to reuse the plic flag, since only the helpers need to discern between models. I'll get it done right now.